packages/bot-web-ui/src/stores/download-store.js

```diff
@@ -0,0 +1,109 @@
+                return localize('Loss amount: {{profit}}', { profit });
+            }
+            default:
+                return null;
+        }
+    };
+
+    @action.bound
+    onDownloadClick = () => {
+        // Transaction Array
+        const rowsTransaction = [
+            [
+                'Market',
+                'Reference ID (buy)',
+                'Reference ID (sell)',
+                'Barrier',
+                'Start Time',
+                'Entry Spot',
+                'Entry Spot Time',
+                'Exit Spot',
+                'Exit Spot Time',
+                'Buy Price',
+                'Profit/Loss',
+            ],
+        ];
+        const transactionsMessages = this.root_store.transactions.elements;
+        transactionsMessages.map(item => {
+            const arr = [
+                item.data.display_name,
+                item.data.transaction_ids.buy,
+                item.data.transaction_ids.sell,
+                item.data.barrier,
+                item.data.date_start,
+                item.data.entry_tick,
+                item.data.entry_tick_time,
+                item.data.exit_tick,
+                item.data.exit_tick_time,
+                item.data.buy_price,
+                item.data.profit,
+            ];
+            rowsTransaction.push(arr);
+        });
+
+        // Journal Array
+        const rowsJournal = [['Date', 'Time', 'Message']];
+
+        const journalMessages = this.root_store.journal.filtered_messages;
+
+        journalMessages.map(item => {
+            let arrayMessage;
+            if (item.message_type !== 'success') {
+                arrayMessage = item.message;
+            } else {
+                arrayMessage = this.getSuccessJournalMessage(item.message.toString().toUpperCase(), item.extra);
+            }
+
+            const arr = [item.date, item.time, arrayMessage];
+            rowsJournal.push(arr);
+        });
+
+        // Push Transaction array to CSV
+        const transactionCsvContent = `data:text/csv;charset=utf-8, ${rowsTransaction
+            .map(e => e.join(','))
+            .join('\n')}`;
+        const transactionEncodedUri = encodeURI(transactionCsvContent);
+        const transactionLink = document.createElement('a');
+        transactionLink.setAttribute('href', transactionEncodedUri);
+        transactionLink.setAttribute('download', 'Transactions.csv');
+        document.body.appendChild(transactionLink);
+
+        // Push Journal array to CSV
+        const journalCsvContent = `data:text/csv;charset=utf-8, ${rowsJournal.map(e => e.join(',')).join('\n')}`;
+        const journalEncodedUri = encodeURI(journalCsvContent);
+        const journalLink = document.createElement('a');
+        journalLink.setAttribute('href', journalEncodedUri);
+        journalLink.setAttribute('download', 'Journal.csv');
+        document.body.appendChild(journalLink);
+
+        transactionLink.click();
+        journalLink.click();
```

Comment:
Should we remove these elements after the files are downloaded?

Reply:
That won't be required since it maps the content in the tabs and is pushed to empty arrays. It allows files to be downloaded as long as the button is enabled.